alf/algorithms/agent.py

```diff
@@ -415,6 +415,17 @@ def preprocess_experience(self, root_inputs, rollout_info, batch_info):
 
         exp, rl_info = self._rl_algorithm.preprocess_experience(
             exp, rollout_info.rl, batch_info)
+
+        # Add batch_info into rollout_info to populate discounted_return.
+        if hasattr(rl_info,
+                   "batch_info") and batch_info.discounted_return != ():
```

Comment:
how about just hasattr(rl_info, "discounted_return") and only pass discounted_return to rl_info to simply code?

Reply:
Isn't HER relabeling a DataTransformer? It can populate rollout_info by itself.